### DISCUSSION TITLE
Bootstrapping in detail

### DISCUSSION THREAD

**Discussion by saddy001** [edtechre/pybroker#70] [Q&A] [answered]
At first, thank you for this library.

The docs say, that the [bias corrected and accelerated (BCa) bootstrap method](https://blogs.sas.com/content/iml/2017/07/12/bootstrap-bca-interval.html) is used. However, this method was designed for data that is independent and identically distributed (iid). As this is not the case for stock data, I wonder how exactly pybroker applies bootstrapping.

If I understand BCa correctly, it draws samples randomly with a specific sample size, as does pybroker. However, because the data is assumed independent in BCa, it draws _each_ sample randomly. From data in `range(10)` with a sample size of 3 it may choose `[7, 2, 4]`. In contrast, pybroker should always choose consecutive samples like `[3, 4, 5]`. Is this the case? I could not grasp that by looking at _bca_boot_conf()_.

**Comment by edtechre** [comment]
Hi @saddy001,

No, Pybroker does not choose consecutive samples when bootstrapping. However, I can add an option to support this.

**Reply by saddy001** [reply]
If it does not, I cannot imagine how the evaluation is made. Can you give a simple example what samples may be used for training and which ones for testing? Because I thought a trading strategy can only be applied to consecutive samples that are ordered by time and have equally sized time gaps between them. What sense makes a moving average, for example, if it is applied to random samples?

**Reply by edtechre** [reply]
Consecutive data is used for train and test splits. The BCa calculation uses random samples.

**Reply by saddy001** [reply]
If I understand you correctly, the samples _within_ a certain split are ordered by time, and a test split directly follows a train split. Only the ordering of the splits itself is sampled randomly. If this is the case, everything is fine as it is.

**Reply by edtechre** [reply, marked as answer]
I am not sure what you mean by "the ordering of the splits itself is sampled randomly."  The BCa is sampled randomly from the test split(s). 

The data in the train and test window is always ordered by date, you can verify this yourself in your execution function. BCa samples randomly to calculate confidence intervals.
